modules/nf-core/hifiasm/main.nf

```diff
@@ -4,15 +4,11 @@ process HIFIASM {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/hifiasm:0.19.9--h43eeafb_0' :
-        'biocontainers/hifiasm:0.19.9--h43eeafb_0' }"
+        'https://depot.galaxyproject.org/singularity/hifiasm:0.20.0--h43eeafb_0' :
+        'biocontainers/hifiasm:0.20.0--h43eeafb_0' }"
 
     input:
-    tuple val(meta), path(reads)
-    path  paternal_kmer_dump
-    path  maternal_kmer_dump
-    path  hic_read1
-    path  hic_read2
+    tuple val(meta), path(reads), path(paternal_kmer_dump), path(maternal_kmer_dump), path(hic_read1), path(hic_read2)
```

Comment:
Some inputs are optional, some must co-exist, and some are mutually exclusive. Also some inputs are likely to be the same specimen as `reads`, others not.
I think it makes more sense to group maternal+paternal and the two hic like this:
```suggestion
    tuple val(meta), path(reads)
    tuple path(paternal_kmer_dump), path(maternal_kmer_dump)
    tuple path(hic_read1), path(hic_read2)
```
(I would in fact like seeing a meta map `val(metak)` added for the kmer_dump inputs, and one named `val(metah)` for the hic inputs. Even though they're not used by the module script code, having the meta maps available allows people to use their values to set resources (cpu, memory, etc) dynamically and tends to simplify code by not having to have a `map` that removes the meta map.

Reply:
After seeing some internal conversations today, Matthieu's comments are consistent with what we're seeing too. 
However Daniels point is that these inputs should be synchronised at some point if one is running multiple instances of hifiasm in a workflow ( which is really the same for any process input ). There was some discussion on using `multiMap` yesterday in the #team-maintainers channel, but Ben confirmed that the way we're using it is ok. 
I think I'm leaning towards this (Matthieu's) suggestion for grouping the inputs as not everyone has or needs these other inputs. I'm not going to comment on the coding ease as I have expert blindness now.

Reply:
I see your point. I'll change it accordingly! 👍